### PR TITLE
LPS-133399 | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-133399
cc @Tim-Cao

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 1]
```diff
@@ -1710,4 +1710,50 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-124104. View the page editor is available with custom master page after staging is enable."
+	@priority = "4"
+	test ViewPageEditorIsAvailableInCustomMasterPageAfterStagingEnabled {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1715]: ViewPageEditorOfContentPageBasedOnCustomMasterInStagingSite

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 2]
```diff
@@ -1710,4 +1710,50 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-124104. View the page editor is available with custom master page after staging is enable."
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1713]: The page editor of content page based on custom master should be available in staging site.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 3]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+	@priority = "4"
+	test ViewPageEditorIsAvailableInCustomMasterPageAfterStagingEnabled {
+		task ("Add a Heading fragment to custom masters") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1716]: Add a Heading fragment to custom master

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 4]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+			MastersPageTemplatesAdmin.gotoMasters(masterLayoutName = "Custom Masters Page Template");
+
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				dropTarget = "Header",
+				fragmentName = "Heading",
+				scope = "Master");
+
+			Button.click(button = "Publish Master");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1726]: Add step to view success message

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 5]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+
+		task ("Add a page based on custom masters") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1728]: Add a content page based on custom master

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 6]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+				pageName = "Test Page Name");
+
+			PageEditor.clickPublish();
+		}
+
+		task ("Activate local live staging") {
+			Staging.openStagingAdmin(siteURLKey = "test-site-name");
+
+			Staging.activateStagingCP(siteName = "Test Site Name");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1742]: Unneeded

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 7]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+
+		task ("Edit the Content Page in staging site") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1744]: Navigate to the content page at edit mode in staging site

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 8]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+			SitePages.openPagesAdmin(siteURLKey = "test-site-name-staging");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1745]: Unneeded

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 9]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+
+			ContentPagesNavigator.openEditContentPage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name Staging");
+		}
+
+		task ("Assert the Heading fragment is shown with staging") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1752]: View the Heading fragment is shown on header

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 10]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+			AssertVisible(
+				key_fragmentName = "heading",
+				locator1 = "Fragment#FRAGMENT_BASIC_COMPONENT");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1755]:
```suggestion
			AssertVisible(
				key_fragmentName = "heading",
				locator1 = "PageEditor#FRAGMENT_INHERITED_FROM_CUSTOM_MASTERS");
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 11]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+		}
+	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1757]: Add step to assert sidebar loaded

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase [hunk 12]
```diff
@@ -1710,4 +1710,50 @@ definition {
...
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/MastersPageTemplates.testcase:1758]: Add step to add a fragment to page then view the fragment is shown

**Comment by Tim-Cao** [comment]
Hi @ozysouza, Could you move the case to MasterPageTemplatesWithStaging.testcase

**Comment by ozysouza** [comment]
Hi @Tim-Cao PR updated following your comments.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/staging/MasterPageTemplatesWithStaging.testcase [hunk 13]
```diff
@@ -267,4 +267,56 @@ definition {
...
+		task ("Add a content page based on custom master") {
+			SitePages.openPagesAdmin(siteURLKey = "test-site-name-staging");
+
+			ContentPages.addPage(
+				masterLayoutName = "Custom Masters Page Template",
+				pageName = "Test Page Name Staging");
+
+			PageEditor.clickPublish();
+
+			Alert.viewSuccessMessage();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/staging/MasterPageTemplatesWithStaging.testcase:292]: Redundant step

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/staging/MasterPageTemplatesWithStaging.testcase [hunk 14]
```diff
@@ -267,4 +267,56 @@ definition {
...
+	test ViewPageEditorOfContentPageBasedOnCustomMasterInStagingSite {
+		task ("Add a Heading fragment to custom master") {
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				dropTarget = "Header",
+				fragmentName = "Heading",
+				scope = "Master");
+
+			Button.click(button = "Publish Master");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/staging/MasterPageTemplatesWithStaging.testcase:281]: Add step Alert.viewSuccessMessage();

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/staging/MasterPageTemplatesWithStaging.testcase [hunk 15]
```diff
@@ -267,4 +267,56 @@ definition {
...
+				pageName = "Test Page Name",
+				siteName = "Test Site Name Staging");
+		}
+
+		task ("View the Heading fragment is shown on header") {
+			AssertVisible(
+				key_fragmentName = "heading",
+				locator1 = "PageEditor#FRAGMENT_INHERITED_FROM_CUSTOM_MASTERS");
+
+			PageEditor.assertFragmentSidebarLoaded();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/staging/MasterPageTemplatesWithStaging.testcase:306]: Add task to view fragment sidebar shown

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/staging/MasterPageTemplatesWithStaging.testcase [hunk 16]
```diff
@@ -267,4 +267,56 @@ definition {
...
+		}
+
+		task ("Add a Button fragment to page") {
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				fragmentName = "Button");
+		}
+
+		task ("View the Button fragment on the page") {
+			AssertVisible(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/masterspagetemplate/staging/MasterPageTemplatesWithStaging.testcase:316]: View Go Somewhere shown

**Comment by Tim-Cao** [comment]
See changes https://github.com/liferay-echo/liferay-portal/compare/0d74543c4684c7aee6e84b255f96341d424ff229...08591a09aa5b927208994cc4287edeb65dbee612
https://github.com/liferay-echo/liferay-portal/pull/4961

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**